app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1705,68 +1705,56 @@ public void openDownloadDialog() {
-                binding.detailPositionView.setVisibility(View.GONE);
-                // TODO: Remove this check when separation of concerns is done.
-                //  (live streams weren't getting updated because they are mixed)
-                if (!info.getStreamType().equals(StreamType.LIVE_STREAM)
-                        && !info.getStreamType().equals(StreamType.AUDIO_LIVE_STREAM)) {
-                    return;
-                }
-            } else {
-                // Show saved position from backStack if user allows it
-                showPlaybackProgress(playQueue.getItem().getRecoveryPosition(),
-                        playQueue.getItem().getDuration() * 1000);
-                animate(binding.positionView, true, 500);
-                animate(binding.detailPositionView, true, 500);
-            }
-            return;
-        }
-        final HistoryRecordManager recordManager = new HistoryRecordManager(requireContext());
 
-        // TODO: Separate concerns when updating database data.
-        //  (move the updating part to when the loading happens)
-        positionSubscriber = recordManager.loadStreamState(info)
-                .subscribeOn(Schedulers.io())
-                .onErrorComplete()
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(state -> {
-                    showPlaybackProgress(state.getProgressMillis(), info.getDuration() * 1000);
-                    animate(binding.positionView, true, 500);
-                    animate(binding.detailPositionView, true, 500);
-                }, e -> {
-                    if (DEBUG) {
-                        e.printStackTrace();
-                    }
-                }, () -> {
-                    binding.positionView.setVisibility(View.GONE);
-                    binding.detailPositionView.setVisibility(View.GONE);
-                });
+        if (!playbackResumeEnabled
+                || playQueue == null || playQueue.getStreams().isEmpty()
+                || playQueue.getItem().getRecoveryPosition() == RECOVERY_UNSET) {
+            binding.positionView.setVisibility(View.GONE);
+            binding.detailPositionView.setVisibility(View.GONE);
+        } else {
+            // Show saved position from backStack if available and user allows it
+            final HistoryRecordManager recordManager = new HistoryRecordManager(requireContext());
+
+            positionSubscriber = recordManager.loadStreamState(info)
+                    .subscribeOn(Schedulers.io())
+                    .onErrorComplete()
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(state -> {
+                        updatePlaybackProgress(
+                                state.getProgressMillis(), info.getDuration() * 1000);
+                    }, e -> {
+                        if (DEBUG) {
+                            e.printStackTrace();
```

Comment:
Please use the logger here

Reply:
Thanks! Gonna improve that, here I use the original code.